### ISSUE TITLE
Cloning corrupts map files (branch backend_compression_map)

### ISSUE THREAD

**Issue by mmd-osm** [drolbr/Overpass-API#246]
Testing cloning on branch https://github.com/drolbr/Overpass-API/tree/backend_compression_map with a Greece country extract.

Before cloning:

```
bin/osm3s_query --db-dir=btrfs/greece_zlib/ 
encoding remark: Please enter your query and terminate it with CTRL+D.
way(8673577);out geom;
<?xml version="1.0" encoding="UTF-8"?>
<osm version="0.6" generator="Overpass API">
<note>The data included in this document is from www.openstreetmap.org. The data is made available under ODbL.</note>
<meta osm_base=""/>

  <way id="8673577">
    <bounds minlat="36.4174619" minlon="26.5948753" maxlat="36.4206816" maxlon="26.6015472"/>
    <nd ref="61917241" lat="36.4174619" lon="26.5986946"/>
    <nd ref="61915156" lat="36.4178520" lon="26.5999643"/>
    <nd ref="61913868" lat="36.4183665" lon="26.6001201"/>
    <nd ref="61917814" lat="36.4188840" lon="26.6012306"/>
    <nd ref="61916533" lat="36.4191410" lon="26.6012293"/>
    <nd ref="61920076" lat="36.4191420" lon="26.6015472"/>
    <nd ref="61921536" lat="36.4206816" lon="26.6009035"/>
    <nd ref="61922793" lat="36.4201601" lon="26.5986808"/>
    <nd ref="61924547" lat="36.4196462" lon="26.5986835"/>
    <nd ref="61914576" lat="36.4197711" lon="26.5977290"/>
    <nd ref="61916037" lat="36.4191229" lon="26.5958250"/>
    <nd ref="61917737" lat="36.4183510" lon="26.5955111"/>
    <nd ref="61919076" lat="36.4183487" lon="26.5948753"/>
    <nd ref="61915857" lat="36.4179639" lon="26.5951951"/>
    <nd ref="61917241" lat="36.4174619" lon="26.5986946"/>
    <tag k="converted_by" v="mikes"/>
    <tag k="created_by" v="JOSM"/>
    <tag k="natural" v="coastline"/>
    <tag k="source" v="PGS"/>
  </way>

</osm>


```

Cloning:
```
bin/osm3s_query --db-dir=btrfs/greece_zlib/ --clone=btrfs/greece_zlib2/
```

After Cloning:

```
bin/osm3s_query --db-dir=btrfs/greece_zlib2/ 
encoding remark: Please enter your query and terminate it with CTRL+D.
way(8673577);out geom;
<?xml version="1.0" encoding="UTF-8"?>
<osm version="0.6" generator="Overpass API">
<note>The data included in this document is from www.openstreetmap.org. The data is made available under ODbL.</note>
<meta osm_base=""/>


</osm>
```

`way(8673577);` is usually processed via a lookup on the way map file. That doesn't seem to work any longer. Regression was introduced in the backend_compression_map branch, it does not happen on v0.7.52.

**Comment by mmd-osm** [comment]
See this commit:

https://github.com/mmd-osm/Overpass-API/commit/47cca26726a69c32bda1b038e3fa4f8b3b27a29e

src/overpass_api/osm-backend/clone_database.cc


```C
      {
       if (src_idx.blocks[i].pos != src_idx.npos)
       {
-	for (uint32 j = 0; j < src_idx.get_block_size()/TIndex::max_size_of(); ++j)
+	for (uint32 j = 0; j < src_idx.get_block_size()*src_idx.get_max_size()/TIndex::max_size_of(); ++j)
 	{
 	  TIndex val =
-	      src_file.get(i*(src_idx.get_block_size()/TIndex::max_size_of()) + j);
+	      src_file.get(i*(src_idx.get_block_size()*src_idx.get_max_size()/TIndex::max_size_of()) + j);
 	  if (!(val == TIndex(uint32(0))))
-	    dest_file.put(i*(src_idx.get_block_size()/TIndex::max_size_of()) + j, val);
+	    dest_file.put(i*(src_idx.get_block_size()*src_idx.get_max_size()/TIndex::max_size_of()) + j, val);
 	}
       }
     }
```

**Comment by mmd-osm** [comment]
Reminder: when working on this ticket, please don't forget to revisit: https://github.com/drolbr/Overpass-API/issues/215#issuecomment-237361228

**Comment by drolbr** [comment]
Fixed in ade51fafd2735b92eb90f1075a219edab58e4944

Thank you, mmd, for the patch to cherry-pick from. The code got in addition some polishing because I did not understand why the patch should work in the first try. But everything is fine.

**Comment by mmd-osm** [comment]
@drolbr : please take a look at the comment above again:

Reminder: when working on this ticket, please don't forget to revisit: #215 (comment)

We need the following patch in random_file_index.h, otherwise we'll get some segmentation faults during planet load.

```C
--- a/src/template_db/random_file_index.h
+++ b/src/template_db/random_file_index.h
@@ -210,7 +210,7 @@ inline Random_File_Index::Random_File_Index
        uint32 void_index_size = void_blocks_file.size("Random_File:11");
        Void_Pointer< uint8 > index_buf(void_index_size);
        void_blocks_file.read(index_buf.ptr, void_index_size, "Random_File:15");
-       for (uint32 i = 0; i < void_index_size/sizeof(uint32); ++i)
+       for (uint32 i = 0; i < void_index_size/8; ++i)
          void_blocks.push_back(*(std::pair< uint32, uint32 >*)(index_buf.ptr + 
        empty_index_file_used = true;
       }

```
